src/lib/marlin_plonk_bindings/stubs/src/pasta_fp_plonk_proof.rs

```diff
@@ -63,18 +63,18 @@ pub fn caml_pasta_fp_plonk_proof_create(
     let auxiliary_input: &Vec<Fp> = &*auxiliary_input;
     let index: &DlogIndex<GAffine> = &index.as_ref().0;
 
-    ocaml::runtime::release_lock();
-
-    let map = GroupMap::<Fq>::setup();
-    let proof = DlogProof::create::<
-        DefaultFqSponge<VestaParameters, PlonkSpongeConstants>,
-        DefaultFrSponge<Fp, PlonkSpongeConstants>,
-    >(&map, auxiliary_input, index, prev)
-    .unwrap();
-
-    ocaml::runtime::acquire_lock();
-
-    proof
+    // NB: This method is designed only to be used by tests. However, since creating a new reference will cause `drop` to be called on it once we are done with it. Since `drop` calls `caml_shutdown` internally, we *really, really* do not want to do this, but we have no other way to get at the active runtime.
```

Comment:
what do you mean by "creating a new reference" here? I'm not sure I see this in the code.

Also, what is caml_shutdown?

I'm not sure I really understand the implications of doing this, especially as [the doc](https://docs.rs/ocaml/0.22.0/ocaml/struct.Runtime.html#safety) is pretty clear that this shouldn't be used outside of test code.

Is there an alternative here?